aten/src/ATen/core/Generator.h

```diff
@@ -1,24 +1,62 @@
 #pragma once
 
-#include <ATen/core/ATenGeneral.h>
 #include <stdint.h>
+#include <memory>
+#include <mutex>
+#include <deque>
+#include <atomic>
+#include <typeinfo>
+#include <utility>
+
+#include <c10/util/Exception.h>
+#include <c10/util/C++17.h>
+#include <ATen/Device.h>
+
+/*
+* Generator note.
+* A Pseudo Random Number Generator (PRNG) is an engine that uses an algorithm to 
+* generate a seemingly random sequence of numbers, that may be later be used in creating 
+* a random distribution. Such an engine almost always maintains a state and requires a
+* seed to start off the creation of random numbers. Often times, users have
+* encountered that it could be beneficial to be able to create, retain, and destroy 
+* PRNG states and also be able to have control over the seed value.
+*
+* A Generator in ATen gives users the ability to read, write and modify a PRNG engine.
+* For instance, it does so by letting users seed a PRNG engine, fork the state of the
+* engine, etc.
+*
+* By default, there is one generator per device, and a device's generator is 
+* lazily created. A user can use the torch.Generator() api to create their own generator.
+* Currently torch.Generator() can only create a CPUGenerator.
```

Comment:
Nice docs! Thank you!

Reply:
This class is CPU only in this PR, is that right?

Reply:
the torch.Generator() api is CPU only in this PR (also it's cpu only in master as well):
https://github.com/pytorch/pytorch/blob/2b826c6171608bebd96be6dd1f8201ab6a0a8dff/torch/csrc/Generator.cpp#L47-L69
In the second PR, I want to be able to create CUDAGenerator when I pass the device="cuda" in torch.Generator()